src/main/java/Duke.java

```diff
@@ -1,10 +1,138 @@
+import java.util.Scanner;
+import java.util.ArrayList;
 public class Duke {
+    private static final String LINE = "____________________________________________________________";
+    private static final String BLANK = "    ";
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printWelcome();
+        Scanner in = new Scanner(System.in);
+        String inputString;
+        ArrayList<Task> taskList = new ArrayList<>();
+        int numberOfTasks = 0;
+
+        while (true) {
+            try {
+                inputString = in.nextLine();
+                String[] command = inputString.split(" ", 2);
+
+                switch (command[0]) {
+                case "list":
+                    doList(taskList, numberOfTasks);
+                    break;
+
+                case "bye":
+                    doExit();
+                    return;
+
+                case "delete":
+                    deleteTask(taskList, Integer.parseInt(command[1]) - 1, numberOfTasks);
+                    numberOfTasks -= 1;
+                    break;
+
+                //mark/unmark command
+                case "mark":
+                case "unmark":
+                    doMarkOrUnmarked(taskList, numberOfTasks, command);
+                    break;
+
+                //add task to list
+                case "todo":
+                    addTodo(taskList, numberOfTasks, command);
+                    numberOfTasks += 1;
+                    break;
+
+                case "deadline":
+                    addDeadline(taskList, numberOfTasks, command);
+                    numberOfTasks += 1;
+                    break;
+
+                case "event":
+                    addEvent(taskList, numberOfTasks, command);
+                    numberOfTasks += 1;
+                    break;
+
+                default:
+                    System.out.println("Unknown command issued");
+                }
+            }
+            catch (IndexOutOfBoundsException e) {
+                System.out.println("Wrong index");
+            }
+            catch (NumberFormatException e) {
+                System.out.println("Number only for argument");
```

Comment:
Can we override the getMessage() method of Exceptions to print error messages instead?